components/kyma-environment-broker/internal/process/deprovisioning/avs_evaluations.go

```diff
@@ -37,9 +37,15 @@ func (ars *AvsEvaluationRemovalStep) Name() string {
 
 func (ars *AvsEvaluationRemovalStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
 	logger.Infof("Avs lifecycle %+v", operation.Avs)
-	if operation.Avs.AVSExternalEvaluationDeleted && operation.Avs.AVSInternalEvaluationDeleted {
-		logger.Infof("Both internal and external evaluations have been deleted")
-		return operation, 0, nil
+	if operation.Avs.AVSInternalEvaluationDeleted {
```

Comment:
How about two functions:

1. logic for trial or freemium
2. logic for remaining plans

 Wouldn't it be more readable and bring us closer to extraction on step level (possibly in the future).

Reply:
Or much simpler.
Each call of DeleteAVSEvaluation is guarded with flag checking.
We wouldn't increase the complexity, nesting level and we would avoid code duplication which is present in your solution.